File: backend/src/main/kotlin/com/mongle/controller/SeedController.kt
package com.mongle.controller

import com.mongle.common.context.AuthUser
import com.mongle.common.context.UserPrincipal
import com.mongle.service.DemoDataSeeder
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@Tag(name = "시드", description = "현재 사용자의 데모 데이터를 준비한다.")
@RestController
@RequestMapping("/api/v1/seed")
class SeedController(
    private val demoDataSeeder: DemoDataSeeder,
) {
    @Operation(summary = "현재 사용자 데모 데이터 시드")
    @PostMapping
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun seed(@AuthUser user: UserPrincipal) {
        demoDataSeeder.seed(user.id)
    }
}

File: backend/src/seed/seed.controller.ts
import { Controller, HttpCode, HttpStatus, Post, UseGuards } from '@nestjs/common';
import { ApiBearerAuth, ApiNoContentResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
import { CurrentUser } from '../common/auth/current-user.decorator';
import { JwtAuthGuard } from '../common/auth/jwt-auth.guard';
import { UserPrincipal } from '../common/auth/user-principal.interface';
import { uuidToBytes } from '../common/prisma';
import { SeedService } from './seed.service';

@ApiTags('시드')
@ApiBearerAuth('bearerAuth')
@UseGuards(JwtAuthGuard)
@Controller('api/v1/seed')
export class SeedController {
  constructor(private readonly seedService: SeedService) {}

  @Post()
  @HttpCode(HttpStatus.NO_CONTENT)
  @ApiOperation({
    operationId: 'seed',
    summary: '현재 사용자 데모 데이터 시드',
  })
  @ApiNoContentResponse({ description: '시드 완료 또는 이미 완료됨.' })
  seed(@CurrentUser() user: UserPrincipal): Promise<void> {
    return this.seedService.seed(uuidToBytes(user.id));
  }
}
